sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java

```diff
@@ -0,0 +1,354 @@
+     *
+     * @param policy The retry policy for service requests.
+     * @return The updated TextAnalyticsClientBuilder object.
+     * @throws NullPointerException If {@code policy} is {@code null}.
+     */
+    public TextAnalyticsClientBuilder addPolicy(HttpPipelinePolicy policy) {
+        Objects.requireNonNull(policy);
+        policies.add(policy);
+        return this;
+    }
+
+    /**
+     * Sets the HTTP client to use for sending and receiving requests to and from the service.
+     *
+     * @param client The HTTP client to use for requests.
+     * @return The updated TextAnalyticsClientBuilder object.
+     */
+    public TextAnalyticsClientBuilder httpClient(HttpClient client) {
+        if (this.httpClient != null && client == null) {
+            logger.info("HttpClient is being set to 'null' when it was previously configured.");
+        }
+
+        this.httpClient = client;
+        return this;
+    }
+
+    /**
+     * Sets the HTTP pipeline to use for the service client.
+     *
+     * If {@code pipeline} is set, all other settings are ignored, aside from
+     * {@link TextAnalyticsClientBuilder#endpoint(String) endpoint} to build {@link TextAnalyticsAsyncClient} or {@link
+     * TextAnalyticsClient}.
+     *
+     * @param httpPipeline The HTTP pipeline to use for sending service requests and receiving responses.
+     * @return The updated TextAnalyticsClientBuilder object.
+     */
+    public TextAnalyticsClientBuilder pipeline(HttpPipeline httpPipeline) {
+        if (this.httpPipeline != null && httpPipeline == null) {
+            logger.info("HttpPipeline is being set to 'null' when it was previously configured.");
+        }
+
+        this.httpPipeline = httpPipeline;
+        return this;
+    }
+
+    /**
+     * Sets the configuration store that is used during construction of the service client.
+     *
+     * The default configuration store is a clone of the {@link Configuration#getGlobalConfiguration() global
+     * configuration store}, use {@link Configuration#NONE} to bypass using configuration settings during construction.
+     *
+     * @param configuration The configuration store used to
+     * @return The updated TextAnalyticsClientBuilder object.
+     */
+    public TextAnalyticsClientBuilder configuration(Configuration configuration) {
+        this.configuration = configuration;
+        return this;
+    }
+
+    /**
+     * Sets the {@link HttpPipelinePolicy} that is used when each request is sent.
+     *
+     * The default retry policy will be used if not provided {@link TextAnalyticsClientBuilder#buildAsyncClient()}
+     * to build {@link TextAnalyticsAsyncClient} or {@link TextAnalyticsClient}.
+     * @param retryPolicy user's retry policy applied to each request.
+     * @return The updated TextAnalyticsClientBuilder object.
+     */
+    public TextAnalyticsClientBuilder retryPolicy(HttpPipelinePolicy retryPolicy) {
```

Comment:
Why do we accept a generic `HttpPipelinePolicy` instead of restricting it to a `RetryPolicy`?

Reply:
are we restricting `retryPolicy` to only accept `RetryPolicy`, and if user want to use CustomizedRetryPolicy then they have to call `addPolicy`?

Reply:
I would recommend that they extend `RetryPolicy` than beginning with `HttpPipelinePolicy` as the base class. Right now this leaves a chance for someone to put any policy in here which could result in no `RetryPolicy` being applied, instead whatever policy submitted being applied.